### DISCUSSION TITLE
Is anyone having success on installing this? (Ubuntu 24 & 22 VM)

### DISCUSSION THREAD

**Discussion by deltaexray** [deepdoctection/deepdoctection#350] [Q&A]
Hello everyone, I recently stumbled across DDT and by now I'm seriously asking myself, what the main issue is. 

To put it simple: Has anyone had success in recent days/weeks of installing deepdoctection? I've tried 10 different methods and so but nothing works, hardly, I can't even get the GPU to work let alone ddt itself. Sure, and that's probably it, the error can be on my side but by now I#m frustrated by it more than I'm intrigued by it. 

NOTE: I'm only talking about the install method with a GPU and Tensorflow. 
Note 2: In the meantime, I've tried both the GPU depended way of using it as well as the CPU only way of doing it - Both are not working after spending roughly 30 Hours on this. It doesn't really matter what kind of versions I use, they all are not working in the end. 

So if anyone here is up for a discussion or help, we can have one. 

Cheers, Alex'

**Comment by JaMe76** [comment]
Sorry about that. The aim of this library is to help with extraction and not getting frustrated. 😩 

 I am highly interested to ship a working package. 
 
One issue that I discovered when pushing the latest changes is the  release of numpy 2.0. The dependencies must be modified to exclude this version.

I did not have a chance to fix it yet but will do it asap. 

Apart from that, can you paste an Error Traceback to see what is going wrong?

**Reply by deltaexray** [reply]
Hello, Sorry about the late reply - work got in the way of spending any more time here. 

I'm sorry, I can't provide you with any Traceback since the VM this was running on, at this time, has been redone cause of a crash. However I can add a little more to the table: 
Since my last try to get deepdoctection to work, I've spent a little more time studying the manual or the instructions to install it, and tried the virtual environment method: This work a little better, however I can't get tensorflow to work with it. It may be because of my lack in knowledge or something not working entirely. 
ALL of the testing and trying has been done, so far, in the standard terminal of Ubuntu Desktop 24.04. I know, 22.04 would probably be working better but I can't imagine that this would be the long term solution anyway, to put it on 22.04.
As said I tried the virtual environment way of installing it and it worked a bit more but when it came to run the test commands, it failed and never outputted any of the "it should output something like this" quotes. 

Fun Fact since u mentioned the Numpy 2.0 release: I was quite surprise to see it has been pulled automatically when I updated everything but that's actually very good to know. 

I'll keep you all posted. 
Cheers and Thank you :)
Alex'

**Reply by deltaexray** [reply]
Update: 2

As of now, Trying to install the necessary drivers onto Ubuntu 24.04 Desktop is impossible. CUDA and cuDNN is not working properly and only wants to be version 12 and above. Same for the Driver, which is always 535.

**Reply by deltaexray** [reply]
[First Install Error.txt](https://github.com/user-attachments/files/16041764/First.Install.Error.txt)


This is a, almost 99,5%, replication of my errors last time. So this is it, as of now because I'm not getting any further

**Comment by JaMe76** [comment]
It looks that the installation error results from an issue when trying to build Pyyaml with cython 3.0 (e.g. https://github.com/yaml/pyyaml/issues/724).

**Comment by deltaexray** [comment]
Long time no answer here, I'm sorry it took me so long to respond but work got in the way of focusing on this one here. 

Besides, I haven't focused on figuring this issue out yet. 

But I've got one question: Is there a way of contacting you directly to ask some questions regarding implementing certain features ? I've got some burning asks on my side. 

Cheers :)
